File: parser/src/main/scala/edu/vsu/florab/migrations/scripts/M000InitialMigration.scala
package edu.vsu.florab.migrations.scripts

import edu.vsu.florab.migrations.process.{Metadata, MetadataCollection, MigrationScript, MigrationUnit}

class M000InitialMigration extends MigrationScript {
  override val n: Int = 0

  val initial: MigrationUnit = (metadataCollection: MetadataCollection, migrationData: Metadata) => {
    throw new RuntimeException("initial migration is required")
  }

  override protected val scripts: List[MigrationUnit] =
    initial :: Nil

}

File: migrator/src/main/kotlin/edu/vsu/flora/migrator/scripts/M000InitialMigration.kt
package edu.vsu.flora.migrator.scripts

import edu.vsu.flora.migrator.process.Migration
import edu.vsu.flora.migrator.schema.MigrationCollection
import edu.vsu.flora.migrator.schema.MigrationDescription

class M000InitialMigration(migrationCollection: MigrationCollection) : Migration(
    migrationCollection, 0, listOf(initialMigration)
) {
    companion object {
        val initialMigration = MigrationDescription("initial migration") { _, _ ->
            throw RuntimeException("Should not be invoked if initial migration exists")
        }
    }
}
